packages/backend-defaults/src/entrypoints/httpRouter/httpRouterServiceFactory.ts

```diff
@@ -56,7 +60,17 @@ export const httpRouterServiceFactory = createServiceFactory({
       config,
     });
 
+    let startupRequestPauseTimeout: HumanDuration | undefined;
+    if (config.has('backend.lifecycle.startupRequestPauseTimeout')) {
+      startupRequestPauseTimeout = readDurationFromConfig(config, {
+        key: 'backend.lifecycle.startupRequestPauseTimeout',
+      });
+    }
```

Comment:
We try to keep the factory function clean, let's move this config reading to happen within `createLifecycleMiddleware` instead

Reply:
I just updated the logic. I had to change the middleware signature API, as it was exported from the backend defaults package.